Nullable DATETIME column does not store NULL

bsatarupa opened the pull request:
### Description

Before Fix:
Nullable datetime column was storing default datetime value '1900-01-01 *' instead of NULL

After Fix:
Nullable datetime column is returning NULL value for all datetime related datatypes. Fix includes dropping default stored value for new tables on these datatypes and setting typdefault to null on upgrade-script for existing tables.

Task: BABEL-2769
Signed-off-by: Satarupa Biswas <satarupb@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

surendravishnoi commented on contrib/babelfishpg_common/sql/upgrades/babelfishpg_common--2.1.0--2.2.0.sql:
```diff
@@ -18,5 +18,17 @@ RETURNS sys.nchar
 AS 'babelfishpg_common', 'nchar'
 LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
+-- [BABEL-2769] Nullable DATETIME column does not store NULL
```
should not use Jira reference in code

bsatarupa replied:
removed JIRA reference

bsatarupa replied:
corrected

surendravishnoi commented on test/JDBC/input/nullableDatetime.sql:
```diff
@@ -0,0 +1,57 @@
+-- [BABEL-2769] Nullable DATETIME column does not store NULL
+
+CREATE DATABASE db_babel_2769;
```
no need to use a separate database instead create objects with filename_xx as name, for example nullableDatetime_smalldatetime and so on

bsatarupa replied:
creating all the tables in master db instead

bsatarupa replied:
using master db instead

surendravishnoi commented on test/JDBC/upgrade/13_6/preparation/BABEL-2769-vu-prepare.sql:
```diff
@@ -0,0 +1,33 @@
+CREATE DATABASE db_babel_2769;
```
no need to use a separate database instead create objects with filename_xx as name, for example BABEL-2769-vu-prepare and so on

bsatarupa replied:
creating all the tables in master db instead

bsatarupa replied:
using master db instead

Commit:
Nullable DATETIME column does not store NULL

Before Fix:
Nullable datetime column was storing default datetime value '1900-01-01 *' instead of NULL

After Fix:
Nullable datetime column is returning NULL value for all datetime related datatypes. Fix includes dropping default stored value for new tables on these datatypes and setting typdefault to null on upgrade-script for existing tables.

Task: BABEL-2769
Signed-off-by: Satarupa Biswas satarupb@amazon.com
